text/0757-default-modifier-manager.md

```diff
@@ -0,0 +1,276 @@
+  wiggly = (element) => {
+    if (this.numberOfClicks % 2 === 0) {
+      return;
+    }
+
+    // event callback
+    let animate = () => {
+      element.animate({ /* ... */ });
+    }
+
+    element.addEventListener('mousemove', animate);
+
+    // Cleanup
+    return () => element.removeEventListener('mousemove', animate);
+  }
+
+  increment = () => this.numberOfClicks++;
+}
+```
+```hbs
+Num clicks: {{this.numberOfClicks}}
+
+<button
+  type="button"
+  {{this.wiggly}}
+  {{on 'click' this.increment}}
+>
+  Try to click me
+</button>
+```
+
+This demonstrates that a modifier can:
+ - be defined as needed, there is no need for the modifier to live in its own file or a separate folder
+ - semantically encapsulate setup and teardown logic -- the returned function is the teardown logic
+ - have access to the element without leaking DOM details until the component class
+ - react to changes to tracked data consumed within the setup stage of the modifier (`this.numberOfClicks`)
+
+[gh-ember-modifier]: https://github.com/ember-modifier/ember-modifier/
+[gh-render-modifiers]: https://github.com/emberjs/ember-render-modifiers
+[gh-ecgutt]: https://github.com/tracked-tools/ember-could-get-used-to-this
+[rfc-779]:https://github.com/emberjs/rfcs/pull/779
+
+This pairs nicely with the [First-Class Component Templates RFC](rfc-779);
+where modifiers may be defined in the same file as a template-only component.
+
+<details><summary>Example</summary>
+
+<!--
+  If you're reading this RFC non-rendered, ignore the "jsx" language tag.
+  "jsx" is incorrect, and isn't perfect, but "gjs" syntax highlighting
+  hasn't been added to GitHub's highlighting yet.
+-->
+
+This could be an entire component
+
+```jsx
+const draggable = element => {
+  element.addEventListener( /*... */ );
+
+  return () => element.removeEventListener( /* ... */ );
+};
+
+<template>
+  <div {{draggable}}>text content A</div>
+  <div {{draggable}}>text content B</div>
+  <div {{draggable}}>text content C</div>
+  <div {{draggable}}>text content D</div>
+</template>
+```
+_Note: `<template>` is experimental syntax, and should not be be the focus of this example_
+
+
+</details>
+
+
+## Detailed design
+
+_A Default Manager is not something that can be chosen by the user, but is baked in to the framework
+as a default so that a user doesn't have to build something to use a non-framework-specific variant
+of the three constructs: Helpers, Modifiers, and Components._
+
+The desired usage of a plain function in a template should be:
+ - convenient
+ - reduce boilerplate
+ - be easily portable to JS for developers' mental model of how template and JS interact.
+
+Which results in:
+ - default to positional parameters
+ - all named arguments are grouped into an "options object" as the last parameter.
+   this happens to align with the _syntax_ of modifier invocation where named arguments may not appear
+   before the last positional argument.
+
+#### Example with mixed params
```

Comment:
For all of these examples, I would suggest finding actual modifier use cases to demonstrate them. That makes the case much stronger. The scroll example I gave is a decent one; you might also grab some of the examples from the guides!

Reply:
Done!, thanks!